docs/v1/P4Runtime-Spec.adoc

```diff
@@ -2577,49 +2416,50 @@ is provided by the P4Runtime client, the server must verify that the length of
 control plane through table key expressions, Value Set lookup expressions,
 Register (PSA extern type) value types, etc. Not supporting these more complex
-types can be very limiting. Table [#tab-p4-type-usage] shows the different
+types can be very limiting. <<#tab-p4-type-usage>> shows the different
 P4~16~ types and how they are allowed to be used, as per the P4~16~
 specification.
 
-~ TableFigure { #tab-p4-type-usage; \
-    caption: "P4 Type Usage"; }
-|----------------|-----------------------|--------------|-----------------|
-|                | Container type                                       |||
-|                |-----------------------|--------------|-----------------|
-| Element type   | header                | header_union | struct or tuple |
-+:---------------+:----------------------+:-------------+:----------------+
-| `bit<W>`       | allowed               | error        | allowed         |
-| `int<W>`       | allowed               | error        | allowed         |
-| `varbit<W>`    | allowed               | error        | allowed         |
-| `int`          | error                 | error        | error           |
-| `void`         | error                 | error        | error           |
-| `error`        | error                 | error        | allowed         |
-| `match_kind`   | error                 | error        | error           |
-| `bool`         | error                 | error        | allowed         |
-| `enum`         | allowed[^enum_header] | error        | allowed         |
-| `header`       | error                 | allowed      | allowed         |
-| header stack   | error                 | error        | allowed         |
-| `header_union` | error                 | error        | allowed         |
-| `struct`       | error                 | error        | allowed         |
-| `tuple`        | error                 | error        | allowed         |
-|----------------|-----------------------|--------------|-----------------|
-~
+//looking for a better solution because is horrible
+.P4 Type Usage
+[.center,%autowidth]
+[#tab-p4-type-usage]
+|===
+| *Element type* 3+^| *Container type*
+|                | *header*               | *header_union* | *struct or tuple* 
+| `bit<W>`       | allowed               | error        | allowed         
+| `int<W>`       | allowed               | error        | allowed         
+| `varbit<W>`    | allowed               | error        | allowed         
+| `int`          | error                 | error        | error           
+| `void`         | error                 | error        | error           
+| `error`        | error                 | error        | allowed         
+| `match_kind`   | error                 | error        | error           
+| `bool`         | error                 | error        | allowed         
+| `enum`         | allowed[^enum_header] | error        | allowed         
```

Comment:
This footnote is using old Madoko syntax.  If we change it to AsciiDoc footnote syntax, note that the footnote will _not_ appear at the bottom of the page in PDF, because AsciiDoc instead puts the footnote at the end of the top-level section instead, which could be many pages later than the table.

If you want to get the footnote to appear immediately below the table, one way is to do what I did in the P4 language spec AsciiDoc for the table titled "Type nesting rules", which is _not_ an AsciiDoc markup footnote.